### ISSUE TITLE
Provide multi-class AUC ROC

### ISSUE THREAD

**Issue by mfeurer** [automl/auto-sklearn#1482]
Scikit-learn provides multi-class options for area under curve: https://scikit-learn.org/stable/modules/generated/sklearn.metrics.roc_auc_score.html

We should provide the most common ones, such as the [OVO Macro averaging used by Auto-Gluon](https://github.com/awslabs/autogluon/blob/0b38dde5f698dbadfa1ce76aabda14505d9e3ead/core/src/autogluon/core/metrics/__init__.py#L443).

**Comment by deadly-panda** [comment]
Hey @mfeurer, can I try working on this  ?

**Comment by mfeurer** [comment]
Sure, please go ahead. Please let us know if yo need further guidance.

**Comment by deadly-panda** [comment]
Just to make sure I'm on the right track. What I need to do is :
1. Create the multi_class_roc_auc scorer (same as the auto-gluon link)
2. Add unit tests for the scorer.
3. Add the scorer to the classification metrics constant dictionary: `CLASSIFICATION_METRICS`
4. In the 01_create_commands script.py: remove the check for nb class >2: continue
Is there something else I am missing?

**Comment by mfeurer** [comment]
Hey, I agree, except for the name. AutoGluon has a very specific name (ovo_macro) that describes the exact averaging mechanism, and it would be good if we could be as specific here, too. Also, you can have a look into [our auto-sklearn](https://github.com/automl/auto-sklearn/blob/development/autosklearn/metrics/__init__.py#L420) code to see how we added these for f1, precision, etc.

**Comment by deadly-panda** [comment]
Hi @mfeurer .. Sorry it took too much time, I've been a bit busy.
I have the code ready, however, I still have a few questions.
For the unit tests, am not sure which numbers I should use ? (you can take a look in my branch if u want to see the ones I used).
Also some tests where failling, for example TestMetric:test_classification_multiclass and I had to add the roc_auc metric to the list of skipped metrics, without really understanding why. can you give more details about this?

One last thing, do u think I should open a PR to make it easy for you to see the changes ?

**Comment by shuraih775** [comment]
hi @mfeurer is this issue still assigned to somebody else or it needs a contributer? If yes, then I am happy to contribute.

**Comment by aseemk98** [comment]
Is this still open? I'd like to take a shot at this.

**Comment by rohitsinghnaruka** [comment]
is it still open ?? @mfeurer

**Comment by ak91456** [comment]
is this issue still open ?? i would like to work on it

**Comment by ray-singh** [comment]
Hi @mfeurer , I'd like to be assigned if this issue is still open. Thanks!

**Comment by neelima-bhaktha** [comment]
hello @mfeurer, if this issue is still open i would like to work on it. Thanks!

**Comment by anirudh8926** [comment]
Hi @mfeurer , I believe the issue is still open and I'd like to contribute. Thanks!

**Comment by gulamalikhan52** [comment]
Hi @mfeurer  , I believe the issue is still open and I'd like to contribute. Thanks!
